### PR TITLE
Add through_defaults for RelatedManager methods

### LINKED ISSUE #1919: Missing keyword argument "through_defaults" on ManyRelatedManager methods
# Bug report

## What's wrong

Mypy raises an error when using `through_defaults` on a many to many field manager.

The code works fine on Django 5.0 and is documented here: https://docs.djangoproject.com/en/5.0/topics/db/models/#extra-fields-on-many-to-many-relationships.

Example error:

```
tests/test_assignments.py:93: error: Unexpected keyword argument "through_defaults" for "add" of "ManyRelatedManager" [call-arg]
        cls.members.add(user, through_defaults={'role': ClassMembership.Role.STUDENT})
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Context: `Class` has a `ManyToManyField` relationship with the `User` model through `ClassMembership`, which also has an additional `role` field.

## How is that should be

`ManyRelatedManager` methods `add`, `set` and async equivalents should have an additional keyword argument `through_defaults`.
The type should be `dict[str, Any]`, I suppose.

File: `django-stubs/db/models/fields/related_descriptors.pyi`.

Django docs: https://docs.djangoproject.com/en/5.0/topics/db/models/#extra-fields-on-many-to-many-relationships.

## System information

- OS: MacOS 14.2.1 
- `python` version: 3.12.1
- `django` version: 5.0
- `mypy` version: 1.7.1
- `django-stubs` version: 4.2.7
- `django-stubs-ext` version: 4.2.7

### PR DESCRIPTION
**Pull request by mfosterw**
# I have made things!

## Related issues

Closes #1919. Based on work from #708.

### CODE REVIEW

File: django-stubs/db/models/fields/related_descriptors.pyi [hunk 1]
```diff
@@ -135,12 +135,26 @@ class ManyToManyDescriptor(ReverseManyToOneDescriptor, Generic[_To, _Through]):
 @type_check_only
 class ManyRelatedManager(Manager[_M], Generic[_M]):
     related_val: tuple[int, ...]
-    def add(self, *objs: _M | int, bulk: bool = ...) -> None: ...
-    async def aadd(self, *objs: _M | int, bulk: bool = ...) -> None: ...
+    def add(self, *objs: _M | int, bulk: bool = ..., through_defaults: dict[str, Any] | None = None) -> None: ...
+    async def aadd(self, *objs: _M | int, bulk: bool = ..., through_defaults: dict[str, Any] | None = None) -> None: ...
```
Comment (sobolevn) [inline django-stubs/db/models/fields/related_descriptors.pyi]: We use `...` for defaults.

**Review by sobolevn — COMMENTED** [review]
Please, note that

> Use the through_defaults argument to specify values for the new [intermediate model](https://docs.djangoproject.com/en/5.0/topics/db/models/#intermediary-manytomany) instance(s), if needed. You can use callables as values in the through_defaults dictionary and they will be evaluated once before creating any intermediate instance(s).

**Comment by mfosterw** [comment]
> Please, note that
> 
> > Use the through_defaults argument to specify values for the new [intermediate model](https://docs.djangoproject.com/en/5.0/topics/db/models/#intermediary-manytomany) instance(s), if needed. You can use callables as values in the through_defaults dictionary and they will be evaluated once before creating any intermediate instance(s).

Per https://github.com/typeddjango/django-stubs/pull/708#issuecomment-1569529725,

> they actually mean a callable (Callable[[], Any]) can be used as a value in the through_defaults dictionary.

(i.e. they're covered by `Any`)

**Review by sobolevn — APPROVED** [review]
Thank you!

**Pull request merged**